src/ascii_str.rs

```diff
@@ -427,22 +386,40 @@ impl Error for AsAsciiStrError {
 }
 
 
+#[cfg(not(feature = "no_std"))]
 /// Connvert mutable slices of bytes to `AsciiStr`.
 pub trait AsAsciiStr : AsciiExt {
     /// Convert to an ASCII slice without checking for non-ASCII characters.
     unsafe fn as_ascii_unchecked(&self) -> &AsciiStr;
     /// Convert to an ASCII slice.
     fn as_ascii(&self) -> Result<&AsciiStr,AsAsciiStrError>;
 }
+#[cfg(feature = "no_std")]
+/// Connvert mutable slices of bytes to `AsciiStr`.
+pub trait AsAsciiStr {
+    /// Convert to an ASCII slice without checking for non-ASCII characters.
+    unsafe fn as_ascii_unchecked(&self) -> &AsciiStr;
+    /// Convert to an ASCII slice.
+    fn as_ascii(&self) -> Result<&AsciiStr,AsAsciiStrError>;
```

Comment:
I think it's more precise to name it `as_ascii_str`. When used like `b.as_ascii_str()` you explicitly know you get a `&AsciiStr` which makes the code easier understandable.